app/models/user.rb

```diff
@@ -12,10 +13,31 @@ class User < ApplicationRecord
 
   has_secure_password
 
+  def remember
+    self.remember_token = User.new_token
+    update_attribute :remember_token, User.digest(remember_token)
+  end
+
+  def authenticated? remember_token
+    Bcrypt::Password.new(remember_digest).is_password? remember_token
+  end
+
+  def forget
+    update_attribute :remember_digest, nil
+  end
+
   class << self
```

Comment:
`self` ở line 17 và này có khác nhau ko em

Reply:
em xóa rồi mà nó còn lỡ dính lại ạ , để em thay đổi ạ

Reply:
là sao, chị đang hỏi lý thuyết, xao lại xóa rồi cu

Reply:
dạ , chỗ đó toán tử ba ngôi ạ , nó như nhau ạ .
và hàm này  xử lý chuỗi truyền vào và trả ra một mảng băm .